agent/grpc-middleware/rate.go

```diff
@@ -33,10 +33,15 @@ func ServerRateLimiterMiddleware(limiter RateLimiter, panicHandler recovery.Reco
 			return ctx, status.Error(codes.Internal, "gRPC rate limit middleware unable to read peer")
 		}
 
+		operationType, ok := rpcRateLimitSpecs[info.FullMethodName]
+		if !ok {
+			logger.Warn("failed to determine which rate limit to apply to RPC", "rpc", info.FullMethodName)
+		}
```

Comment:
Should we return here to avoid having to handle the empty string case in Allow?

Reply:
Sounds good 👍🏻